fix: support Fleet 1.35

davfsa opened the pull request:
Mostly a large rewrite of all the theme to work in the new Fleet versions.

I have also fixed some theme issues where text would be hard to read (tweaked opacity) and incorrect colours were being used, as lack of specifying would cause it to fallback to Fleets default colour pallet.

There are also some removed fields as they are no longer supported (mainly per component disable colours, etc)

Any and all feedback is welcome! If you find any inconsistencies in the theme, please do let me know and I'll fix it!

Closes #11 
Alternative to #12

kkrypt0nn commented:
Not sure if your screenshots have some different colors than the actual colors of the application with the theme, but they're quite light compared to how they should be.

| New | Old |
|---|---|
| ![Screenshot_20240608-092217.png](https://github.com/catppuccin/fleet/assets/43011723/140d587c-e6c9-4f3c-adcb-060de8995cb0) | ![Screenshot_20240608-092229.png](https://github.com/catppuccin/fleet/assets/43011723/269c4e7e-2e43-4b9c-b0f1-b35bb8a12b9a) |


If the theme effectively looks like that, there's something off, otherwise the screenshots quality is just low and should be remade in my opinion.

May be worth syncing with #12 as well, and make a single PR together.

Thanks

davfsa commented:
I had a look through #12 and it's effectively the same pr as this one, but it contains a lot of fields that have been removed and do not longer work and still some button rounding errors that just simply porting the new theme to fleet 3.15 would cause.

I also don't know where the lighting changes come from, but I'll investigate!

davfsa commented:
@kkrypt0nn It looks like the tool I was using to convert my screenshot png to webp (cwebp) kept a quality of 80% by default and was also lossy compression by default.

Changed those two things and now the colours should be much more vivid and close to the original picture.

Thanks for pointing it out!

thelooter commented:
> I had a look through #12 and it's effectively the same pr as this one, but it contains a lot of fields that have been removed and do not longer work and still some button rounding errors that just simply porting the new theme to fleet 3.15 would cause.
> 
> I also don't know where the lighting changes come from, but I'll investigate! I havent tweaked any attribute colours, so they should be exactly the same, unless they were made lighter by Fleet.

Is there anything in #12 that isn't in here? Otherwise I would go and close that other one

davfsa commented:
> Is there anything in https://github.com/catppuccin/fleet/pull/12 that isn't in here? Otherwise I would go and close that other one.

From a quick look, it doesn't seem so. We even both included nice new things like the new smartMode theme style!

>  We might also wanna think about adding some blank lines between components so its a bit more maintainable.

Thought about that too, but the original did not seem to have the split lines to I decided not to. Seeing more people also find it hard to maintain due to this makes it actually worth spending the time to clean it up. Will make a commit with it :)

thelooter commented:
> > Is there anything in #12 that isn't in here? Otherwise I would go and close that other one.
> 
> From a quick look, it doesn't seem so. We even both included nice new things like the new smartMode theme style!

Got it, will close my PR then

davfsa commented:
Added spacing.

The only unfortunate side-effect is that the spacing is also added in the generated JSON.

It could be avoided by making the newline comments, but I dislike that approach. Maybe something could be added to whiskers to strip out empty lines?

thelooter commented:
I don't mind having the whitespace in the generated json personally

davfsa commented:
Added some missing button types that I seem to have missed and made the tabs look more like tabs. This means that the selected tab colour will now match the background of the component under it. This is more in line with how the rest of the builtin fleet themes look and I think it adds a nice touch.

Nonetheless, it also deviates a bit from the original theme, so feedback is welcome and ill adjust based on it!

VictorTennekes commented:
Nice work! Looks solid to me.
I think maintaining the exact same look as the original one isn't necessary.
I know this doesn't touch the exact same topic but the spirit of this comment by @backwardspy still goes I'd assume:
> i wouldn't worry too much about going off-palette, the primary aim is for ports to capture catppuccin's look & feel rather than rigidly sticking to the preset colours. shades generated in either of these ways generally meet the criteria!

uncenter commented:
> @kkrypt0nn It looks like the tool I was using to convert my screenshot png to webp (cwebp) kept a quality of 80% by default and was also lossy compression by default.
> 
> Changed those two things and now the colours should be much more vivid and close to the original picture.
> 
> Thanks for pointing it out!

Yeah the solution is to add the `-lossless` flag :)

uncenter commented on assets/frappe.webp:
Not sure if there have been changse since these previews were generated but a lot of the text visible here (filenames in tabs, text in the file tree) appears not to the Catppuccin's `text` color. Additionally, comments don't look to be the correct color either.

davfsa replied:
Really nice spot on both!

The comments seem to have been because you can now style line and bloc comments differently. Just duplicated the same style.

And the filename and tabs text is also fixed. You now have to explicitly set the colour and it no longer falls back to `text.default` (unfortunately)

uncenter replied:
I'm confused on the last part - the preview image still shows the filenames in each tab as the wrong color? I've also noticed the icons and arrows in the file explorer are the wrong color, are they able to be themed?

davfsa replied:
> I'm confused on the last part - the preview image still shows the filenames in each tab as the wrong color?

Yep, sorted that one out in https://github.com/catppuccin/fleet/commit/0fa0073cfd7e039ad884fd0dd406758d0530f0e7. Mishap on my part.

>  I've also noticed the icons and arrows in the file explorer are the wrong color, are they able to be themed?

Would you mind attaching a screenshot on what part you mean exactly? Don't know what you are referring to when you say "icons"

uncenter replied:
![Screenshot 2024-06-08 at 10 08 36 (CleanShot)](https://github.com/catppuccin/fleet/assets/47499684/84df67a5-66f4-40ec-b45a-1e194b938611)

Also the file icons are unthemed - does https://github.com/catppuccin/jetbrains-icons work for Fleet? Might want to install that and take the screenshots with it :)

davfsa replied:
Oh, you mean those!

I'll have a look and get back to you, they should for sure be themed 🤔

> Also the file icons are unthemed - does https://github.com/catppuccin/jetbrains-icons work for Fleet? Might want to install that and take the screenshots with it :)

That I am not too sure I can, as fleet doesnt have plugins support yet

uncenter replied:
> That I am not too sure I can, as fleet doesnt have plugins support yet

No worries then :)

uncenter replied:
Ooh actually this is a little scuffed but it looks like you can load plugins by copying the files maybe? https://stackoverflow.com/a/75148190/17378715

davfsa replied:
After a lot of trying, I have not been able to theme the leftover icons that you marked. They only seem to follow the theme style (dark or light). If someone knows where to even look, please do let me know, as I have tried everywhere.

> Ooh actually this is a little scuffed but it looks like you can load plugins by copying the files maybe? https://stackoverflow.com/a/75148190/17378715

Don't think that will be possible unfortunate. They seem to use completely different SDKs. That comment talked about installing a plugin that is marked as supporting Fleet (as can be seen in the store). Additionally, it is a bundled package now too.

As much as I dont like this, think its better left to something to do in the future, once they release the fleet plugin SDK

thelooter commented on fleet.tera:
`gitBranchTag` seems still unthemed, an example of the options can be found here:
https://github.com/catppuccin/fleet/pull/12/files#diff-864e7a2252f4dcc7c47c8db913bff78a3707331cecf6fcd2e4d8d504224ee5f5R167-R172

davfsa replied:
Missed that!

Should look like this now (https://github.com/catppuccin/fleet/pull/13/commits/bb79e26a69d087fdc72c3d1a1b7c90d9f56e5eef):

![image](https://github.com/catppuccin/fleet/assets/29100934/8cf3c557-8b7c-410a-bdbb-08c73ebb4916)

thelooter reviewed (changes requested):
![image](https://github.com/catppuccin/fleet/assets/24987877/0f4a5e55-74b4-4506-a0c4-8c683faed065)

The Hover on that keyboard button seems either extremly low contrast or doesn't exist.

davfsa commented:
> The Hover on that keyboard button seems either extremly low contrast or doesn't exist.

You are absolutely right, it is way too low contrast when with that background. Im sorry for that!

Switched the hover/pressed colours for a switch to a lighter colour rather than a darker one (https://github.com/catppuccin/fleet/pull/13/commits/5ff993d37e3427c572d523c0f15e047e6912cb5b)
